crates/provider-sqldb-postgres/wit/deps/postgres/types.wit

```diff
@@ -0,0 +1,262 @@
+    numeric-array(list<numeric>),
+
+    serial(u32), serial4(u32),
+
+    small-int(s16), int2(s16),
+    int2-array(list<s16>),
+    int2-vector(list<s16>),
+    int2-vector-array(list<list<s16>>),
+
+    small-serial(s16), serial2(s16), // note: matches tokio-postgres
+
+    // Bytes
+    //
+    // For bit & bit-varying, see the encoding scheme used by bit-vec:
+    // https://contain-rs.github.io/bit-vec/bit_vec/struct.BitVec.html#method.to_bytes
+    bit(tuple<u32, list<u8>>),
+    bit-array(list<tuple<u32, list<u8>>>),
+    bit-varying(tuple<option<u32>, list<u8>>), varbit(tuple<option<u32>, list<u8>>),
+    varbit-array(list<tuple<option<u32>, list<u8>>>),
+    bytea(list<u8>),
+    bytea-array(list<list<u8>>),
+
+    // Characters
+    // TODO: specify text encoding, to negotiate possible component/DB mismatch?
+    %char(tuple<u32, list<u8>>),
+    %char-array(list<tuple<u32, list<u8>>>),
+
+    varchar(tuple<option<u32>, list<u8>>),
+    varchar-array(list<tuple<option<u32>, list<u8>>>),
+
+    // Networking
+    cidr(string),
+    cidr-array(list<string>),
+
+    inet(string),
+    inet-array(list<string>),
+
+    macaddr(tuple<u8, u8, u8, u8, u8, u8>), // EUI-48
+    macaddr-array(list<tuple<u8, u8, u8, u8, u8, u8>>), // EUI-48
+
+    macaddr8(tuple<u8, u8, u8, u8, u8, u8, u8, u8>), // EUI-64 (deprecated)
+    macaddr8-array(list<tuple<u8, u8, u8, u8, u8, u8, u8, u8>>), // EUI-64 (deprecated)
+
+    // Geo
+    box(tuple<lower-left-point, upper-right-point>),
+    box-array(list<tuple<lower-left-point, upper-right-point>>),
+
+    circle(tuple<center-point, radius>),
+    circle-array(list<tuple<center-point, radius>>),
+
+    line(tuple<start-point, end-point>),
+    line-array(list<tuple<start-point, end-point>>),
+
+    lseg(tuple<start-point, end-point>),
+    lseg-array(list<tuple<start-point, end-point>>),
+
+    path(list<point>),
+    path-array(list<list<point>>),
+
+    point(point),
+    point-array(list<point>),
+
+    polygon(list<point>),
+    polygon-array(list<list<point>>),
+
+    // Date-time
+    date(date),
+    date-array(list<date>),
+
+    interval(interval),
+    interval-array(list<interval>),
+
+    time(time),
+    time-array(list<time>),
+
+    time-tz(time-tz),
+    time-tz-array(list<time-tz>),
+
+    timestamp(timestamp),
+    timestamp-array(list<timestamp>),
+
+    timestamp-tz(timestamp-tz),
+    timestamp-tz-array(list<timestamp-tz>),
+
+    // JSON
+    json(string),
+    json-array(list<string>),
+    jsonb(string),
+    jsonb-array(list<string>),
+
+    // Money (use is discouraged)
+    //
+    // fractional precision is determined by the database's `lc_monetary` setting.
+    //
+    // NOTE: if you are storing currency amounts, consider
+    // using integer (whole number) counts of smallest indivisible pieces of currency
+    // (ex. cent amounts to represent United States Dollars; 100 cents = 1 USD)
+    money(numeric),
+    money-array(list<numeric>),
+
+    // Postgres-internal
```

Comment:
Can we get away with eliding these types for now or are we worried about full compatibility?

Reply:
These are useful for people wanting to do replication log things -- we cant actually support them in the *implementation* properly yet, but they should definitely be at the WIT level written down.